File: sample-java/src/main/java/com/mapp/engagesample/CustomFirebaseMessaging.java
package com.mapp.engagesample;

import androidx.annotation.NonNull;

import com.appoxee.Appoxee;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class CustomFirebaseMessaging extends FirebaseMessagingService {
    @Override
    public void onNewToken(@NonNull String token) {
        super.onNewToken(token);
        Appoxee.instance().updateFirebaseToken(token);
    }

    @Override
    public void onMessageReceived(@NonNull RemoteMessage message) {
        super.onMessageReceived(message);
        boolean isMapp= Appoxee.instance().isPushMessageFromMapp(message);
        if(isMapp){
            Appoxee.instance().handlePushMessage(message);
        }else{
            // handle messages for other providers
        }
    }
}

File: shared-ui/src/main/java/eu/brrm/shared_ui/CustomFirebaseMessaging.kt
package eu.brrm.shared_ui

import com.appoxee.Appoxee
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class CustomFirebaseMessaging : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        // update Mapp's token
        Appoxee.instance().updateFirebaseToken(token)

        // update token for other providers
    }

    override fun onMessageReceived(message: RemoteMessage) {
        super.onMessageReceived(message)
        val isMapp = Appoxee.instance().isPushMessageFromMapp(message)
        if (isMapp) {
            Appoxee.instance().handlePushMessage(message)
        } else {
            // handle with some other provider
        }
    }
}